src/data/finset/powerset.lean

```diff
@@ -117,5 +117,37 @@ begin
     exact mem_powerset.mpr (mem_powerset_len.mp ha).1, }
 end
 
+lemma powerset_len_sup [decidable_eq α] (u : finset α) (n : ℕ) (hn : n < u.card) :
```

Comment:
There should be a way to break this down into smaller lemmas. I'm thinking that there could be a `powerset_len_insert` lemma (should be easy to derive from `powerset_insert`) which would make an induction proof possible.

Reply:
@acxxa @pechersky Is this still WIP?

Reply:
I've broken down the lemma as best I could. The set-induction proof ends up being longer, in my attempts. So I think I'm happy with how it is.